kinto_http/batch.py

```diff
@@ -19,10 +19,17 @@ def __init__(self, client, batch_max_requests=0):
         self._results = []
 
     def request(self, method, endpoint, data=None, permissions=None,
-                headers=None):
+                payload=None, headers=None):
         # Store all the requests in a dict, to be read later when .send()
         # is called.
-        self.requests.append((method, endpoint, data, permissions, headers))
+        payload = payload or {}
+        if data is not None:
+            payload['data'] = data
+        if permissions is not None:
+            payload['permissions'] = permissions
+
+        self.requests.append((method, endpoint, payload.get("data"),
```

Comment:
I don't think this will work in case payload doesn't contains data or permissions.
If I recall correctly patch is using the json+patch mimetype reason why they explicitely used payload rather than data and permissions

Reply:
Even with the `_build_requests()` code below?

Reply:
Yes see http://docs.kinto-storage.org/en/stable/api/1.x/records.html#json-patch-operations